psi4/driver/procrouting/sapt/sapt_proc.py

```diff
@@ -70,6 +70,7 @@ def run_sapt_dft(name, **kwargs):
     mon_b_shift = core.get_option("SAPT", "SAPT_DFT_GRAC_SHIFT_B")
     do_delta_hf = core.get_option("SAPT", "SAPT_DFT_DO_DHF")
     sapt_dft_functional = core.get_option("SAPT", "SAPT_DFT_FUNCTIONAL")
+    do_dft = sapt_dft_functional != "HF"
```

Comment:
This is to optimize running SAPT0 with the SAPT(DFT) module by setting DFT functional to HF. In SAPT(DFT) the program does a delta-HF calculation based on HF orbitals and then followed by DFT monomer caluclations; for SAPT0 this will be redundant so I added some logics to avoid this.

Reply:
Could we determine if this is _actually_ a DFT computation or not at the Python layer?

Reply:
I'm not sure what you mean... Do you mean determining if it's a HF or DFT calculation? I think HF and DFT shares the same driver function `run_scf` in Psi4

Reply:
No. I mean that if the user requests to do SAPT(DFT) using Hartree-Fock, the code should direct them to use the normal SAPT(0) code, if that's a more efficient way to do the same computation.

Reply:
Doing a SAPT0 is going to get more complicated before it gets simpler. Right now there are three ways -- Ed, Rob, Daniel, in short. And with some of the more-accurate non-S^2-approx terms that the Patkowski group has implemented, there's good coding reasons that one would run a SAPT0 through the SAPT(DFT) aka "Daniel" route. It makes for unlovely input files to get the highest-level SAPT calc possible from psi. I think Yi's special-casing the HF fctl may be part of figuring out a more rational future for SAPT code routing.

In general, I agree with Jonathon's advice on steering users toward the most efficient route. But here I don't think we want to explain the present or commit to the future. And if users have already strayed this far from `energy("sapt0")`, I think let them continue.

Reply:
That's fine with me, but can you or Yi write an issue that can say what the three ways are, what this means for short-term SAPT development, and (eventually) coordinate plans for improving the situation? This problem is going to come up frequently enough that it deserves a dedicated reference.

Reply:
One thing to consider is, AFAIK, running SAPT0 through the SAPT(DFT) path computes coupled dispersion (orbital relaxation effects) at HF level which none of the alternative routes have implemented.

Reply:
@PhillCli Exactly. One of my next plan is to make coupled dispersion an option so that the code would be more useful with SAPT0, since SAPT0 does not depend on coupled dispersion as heavily as SAPT(DFT) does. As Lori said, one main reason that users would choose to use the "Daniel" code for SAPT0 is that it seems to be the only one that has support to non-S^2-approximated induction & exchange-induction terms.

Reply:
For future reference, the requested notes exist but are not public access. If you need this, contact Lori directly.